tests/config/read.c

```diff
@@ -759,6 +759,25 @@ void test_config_read__bom(void)
 	git_buf_dispose(&buf);
 }
 
+void test_config_read__single_line(void)
+{
+	git_buf buf = GIT_BUF_INIT;
+	git_config *cfg;
+
+	cl_set_cleanup(&clean_test_config, NULL);
+	cl_git_mkfile("./testconfig", "[some] var = value\n[some \"OtheR\"] var = value");
```

Comment:
You might want to have one of these drop the space between the header and the value (e.g., `[some]var = value`) for variety. From my reading of the code, that should be handled just fine, but it's yet another funny corner case.